apple-catalog-parsing/src/asset_catalog.rs

```diff
@@ -0,0 +1,33 @@
+use std::ffi::CString;
+use std::os::unix::ffi::OsStrExt;
+use std::path::Path;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
```

Comment:
[nit] Add `#[non_exhaustive]` to indicate that future versions of the crate could add more variant. This makes it so that adding a new variant is not a breaking API change (as crates which use this must handle the case that some other variant which doesn't exist yet is encountered). 

With this crate not being distributed separately yet, it doesn't matter that much, but good practice in case we do end up distributing it separately.

```suggestion
#[derive(Error, Debug)]
#[non_exhaustive]
pub enum Error {
```